src/content_sao.cpp

```diff
@@ -1208,6 +1215,43 @@ int PlayerSAO::punch(v3f dir,
 
 	if (!damage_handled) {
 		setHP(getHP() - hitparams.hp);
+
+		if (!isDead()) {
+			if (g_settings->getBool("enable_knockback")) {
+				int knockback_power_horizontal = toolcap->knockback_power_horizontal;
```

Comment:
This section is the part that can get a knockback direction from player yaw and view pitch and feed the direction and knockback strength into a more general and separate 'throw player' code. Please remove resistence armour.